tests/epyccel/test_epyccel_functions.py

```diff
@@ -286,10 +286,61 @@ def g(x, y):
     f = epyccel(g)
     assert f(2,1) == g(2,1)
 
+
+def test_decorator_f15():
+    @types('bool', 'int8', 'int16', 'int32', 'int64')
+    def f15(a,b,c,d,e):
+        if a:
+            return b + c
+        else:
+            return d + e
+
+    f = epyccel(f15)
+    assert f(True, 1, 2, 3, 4)  == f15(True, 1, 2, 3, 4)
+    assert f(False, 1, 2, 3, 4)  == f15(False, 1, 2, 3, 4)
+
+
+def test_decorator_f16(language):
+    @types('int16')
+    def f16(a):
+        b = a
+        return b
+    f = epyccel(f16, language=language)
+    assert f(1) == f16(1)
+
+def test_decorator_f17(language):
+    @types('int8')
+    def f17(a):
+        b = a
+        return b
+    f = epyccel(f17, language=language)
+    assert f(1) == f17(1)
+
+def test_decorator_f18(language):
+    @types('int32')
+    def f18(a):
+        b = a
+        return b
+    f = epyccel(f18, language=language)
+    assert f(1) == f18(1)
+
+def test_decorator_f19(language):
+    @types('int64')
+    def f19(a):
+        b = a
+        return b
+    f = epyccel(f19, language=language)
+    assert f(1) == f19(1)
```

Comment:
This should be correct, because we assume that a Python `int` is equivalent to a `numpy.int64`.